diff --git a/dist-js/index.d.ts b/dist-js/index.d.ts
--- a/dist-js/index.d.ts
+++ b/dist-js/index.d.ts
@@ -8,18 +8,18 @@ declare global {
     interface Window {
         __TAURI__?: {
             core?: {
-                invoke?: (cmd: string, args?: InvokeArgs) => Promise<unknown>;
+                invoke?: <T>(cmd: string, args?: InvokeArgs) => Promise<T>;
             };
             log?: {
                 trace?: (message: string) => Promise<void>;
                 debug?: (message: string) => Promise<void>;
                 info?: (message: string) => Promise<void>;
                 warn?: (message: string) => Promise<void>;
                 error?: (message: string) => Promise<void>;
             };
             event?: {
-                listen?: (event: string, callback: (event: {
-                    payload: unknown;
+                listen?: <T>(event: string, callback: (event: {
+                    payload: T;
                 }) => void) => Promise<() => void>;
                 emit?: (event: string, payload: unknown) => Promise<void>;
             };
